### DISCUSSION TITLE
synchronized logging problem

### DISCUSSION THREAD

**Discussion by sadransh** [USC-ACTLab/crazyswarm#322] [Q&A]
Hi, I want to log data from one crazyflie. I used `"rosbag record -a"` to get /tf and other data: I used these settings : 
`genericLogTopics: ["log1","log2"]
 genericLogTopicFrequencies: [10,10]
 genericLogTopic_log1_Variables: [ "gyro.x","gyro.y", "gyro.z"]
 genericLogTopic_log2_Variables: [ "acc.x","acc.y", "acc.z"]`

and the result of rosbag info is like this:
```     tf2_msgs/TFMessage              [94810edda583a504dfda3829e70d7eec]
topics:
     /cf1/log1       230 msgs    : crazyflie_driver/GenericLogData
             /cf1/log2      2156 msgs    : crazyflie_driver/GenericLogData
             /diagnostics     23 msgs    : diagnostic_msgs/DiagnosticArray
             /joy            164 msgs    : sensor_msgs/Joy                
             /rosout          15 msgs    : rosgraph_msgs/Log               (5 connections)
             /rosout_agg       2 msgs    : rosgraph_msgs/Log              
             /tf            1807 msgs    : tf2_msgs/TFMessage'
```
 the collected data are not synchronized. I need to log with somehow high data rate.
I need to know the position of the crazyflie at the moment of collecting data. Is there any way to do so? will using usdcard help to have all the mentioned data? 
I want to log some data related to the motor too. like PWM or thrust for each motor and ..? Do you have any suggestion to log such data?

**Comment by whoenig** [comment]
The values within one log topic are synchronized, but values across different topics are not. You can create your own custom log configuration (as you did for log1 and log2) above and combine the gyro and acc readings. Unfortunately, you can only log ~26 Bytes per log configuration, so about 6 float variables. Finally, logging over the radio is limited (in the firmware) to 100 Hz.

The best choice is likely the uSD card. I have successfully used it to log >20 variables at high rates (you can go up to 1 khz if you only log a few variables). All variables on the uSD card are synchronized (default timestamp in ms, but its also possible to get a us timer). You can download the latest log file from the uSD card over the radio using the crazyflie_tools package.

**Comment by sadransh** [comment]
Great answer! Thank you very much.
How about /tf data and motor are they accessible from firmware to log them into the uSD card?

**Comment by whoenig** [comment]
/tf: It depends on how you localize. I assume that you use a mocap or lighthouse? In this case, the logging variables are defined here: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/stabilizer.c#L636-L677 (the *Z version is compressed to minimize file-size/overhead).

Motor: The raw PWM values have logging variables defined here: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/drivers/src/motors.c#L323-L328.

The uSD card supports all logging variables.

**Comment by jpreiss** [comment]
@sadransh any updates to share with future readers of this issue?

**Comment by easy98** [comment]
Hi, 

I also have the same problem. But since now I have already attached the flow-deck and the lps-deck. So it seems that I cannot add uSD-card deck at the same time. So is there other solutions to solve this problem and get synchronized logging variables? Thanks.

**Comment by whoenig** [comment]
Bitcraze now sells longer pin headers for decks, which should allow you to attach 3 decks. Perhaps that suffices? Otherwise you can use the logging over the Crazyradio (just a few variables can be logged simultaneously), or physically connect a J-Link on the debug port (to collect data without flying).

**Comment by easy98** [comment]
Hi, 

I find the usd-deck seems not compatible with my crazyflie. When I add it, my crazyflie will direct crash on the ground.

**Reply by whoenig** [reply]
Make sure you use the latest firmware. If the issue persists, open a issue with more technical details at https://github.com/bitcraze/crazyflie-firmware.

**Reply by easy98** [reply]
Hi,

I update the firmware. Now the CF can fly normally but the usd card is not able to log anything. You can see this line:

> [ INFO] [1623976496.123899060]: [cf7] uSD: Config read [FAIL].

It's my config.txt file:

> 100   # frequency
> 25    # buffer size
> log   # file name
> 1     # enable on startup (0/1)
> 1     # mode (0: disabled, 1: synchronous stabilizer, 2: asynchronous)
> acc.x
> acc.y
> acc.z
> gyro.x
> gyro.y
> gyro.z
> stateEstimate.x
> stateEstimate.y
> stateEstimate.z

And the results of ros is like this:
> SUMMARY
> ========
> 
> PARAMETERS
>  * /crazyflieTypes/CF21SingleMarker/batteryVoltageWarning: 3.8
>  * /crazyflieTypes/CF21SingleMarker/batteryVoltateCritical: 3.7
>  * /crazyflieTypes/CF21SingleMarker/bigQuad: False
>  * /crazyflieTypes/CF21SingleMarker/dynamicsConfiguration: 0
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_z: 60000
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_z: 12000
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/mass: 0.032
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/massThrust: 132000
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.1
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_xy: 1.0
>  * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_z: 2.0
>  * /crazyflieTypes/CF21SingleMarker/markerConfiguration: 3
>  * /crazyflieTypes/default/batteryVoltageWarning: 3.8
>  * /crazyflieTypes/default/batteryVoltateCritical: 3.7
>  * /crazyflieTypes/default/bigQuad: False
>  * /crazyflieTypes/default/dynamicsConfiguration: 0
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_m_z: 1500
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_xy: 2.0
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_z: 0.4
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_xy: 70000
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_z: 60000
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_omega_rp: 200
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_xy: 0.2
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_z: 0.4
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_m_z: 500
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_xy: 0.05
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_z: 0.05
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_xy: 0.4
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_z: 1.25
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_xy: 20000
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_z: 12000
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/mass: 0.032
>  * /crazyflieTypes/default/firmwareParams/ctrlMel/massThrust: 132000
>  * /crazyflieTypes/default/markerConfiguration: 0
>  * /crazyflieTypes/defaultSingleMarker/batteryVoltageWarning: 3.8
>  * /crazyflieTypes/defaultSingleMarker/batteryVoltateCritical: 3.7
>  * /crazyflieTypes/defaultSingleMarker/bigQuad: False
>  * /crazyflieTypes/defaultSingleMarker/dynamicsConfiguration: 0
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_z: 60000
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_z: 12000
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/mass: 0.032
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/massThrust: 132000
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.2
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_xy: 4.0
>  * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_z: 8.0
>  * /crazyflieTypes/defaultSingleMarker/markerConfiguration: 3
>  * /crazyflieTypes/large/batteryVoltageWarning: 11.4
>  * /crazyflieTypes/large/batteryVoltateCritical: 11.1
>  * /crazyflieTypes/large/bigQuad: True
>  * /crazyflieTypes/large/dynamicsConfiguration: 0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_m_z: 1500
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_xy: 2.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_z: 4.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_xy: 30000
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_z: 40000
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_omega_rp: 100
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_xy: 3.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_z: 3.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_m_z: 500
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_xy: 1.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_z: 2.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_xy: 8.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_z: 8.0
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_xy: 9000
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_z: 10000
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/mass: 0.38
>  * /crazyflieTypes/large/firmwareParams/ctrlMel/massThrust: 5400
>  * /crazyflieTypes/large/markerConfiguration: 2
>  * /crazyflieTypes/medium/batteryVoltageWarning: 7.6
>  * /crazyflieTypes/medium/batteryVoltateCritical: 7.4
>  * /crazyflieTypes/medium/bigQuad: True
>  * /crazyflieTypes/medium/dynamicsConfiguration: 0
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_m_z: 1500
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_xy: 2.0
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_z: 2.0
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_xy: 50000
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_z: 60000
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_omega_rp: 200
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_xy: 1.0
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_z: 1.0
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_m_z: 500
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_xy: 0.1
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_z: 0.5
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_xy: 2.0
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_z: 3
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_xy: 16000
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_z: 12000
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/mass: 0.11
>  * /crazyflieTypes/medium/firmwareParams/ctrlMel/massThrust: 23000
>  * /crazyflieTypes/medium/markerConfiguration: 1
>  * /crazyflies: [{'initialPositio...
>  * /crazyswarm_server/enable_logging: True
>  * /crazyswarm_server/enable_parameters: True
>  * /crazyswarm_server/firmwareParams/commander/enHighLevel: 1
>  * /crazyswarm_server/firmwareParams/kalman/resetEstimation: 1
>  * /crazyswarm_server/firmwareParams/locSrv/extPosStdDev: 1e-3
>  * /crazyswarm_server/firmwareParams/locSrv/extQuatStdDev: 0.05
>  * /crazyswarm_server/firmwareParams/ring/effect: 16
>  * /crazyswarm_server/firmwareParams/ring/headlightEnable: 0
>  * /crazyswarm_server/firmwareParams/ring/solidBlue: 255
>  * /crazyswarm_server/firmwareParams/ring/solidGreen: 0
>  * /crazyswarm_server/firmwareParams/ring/solidRed: 0
>  * /crazyswarm_server/firmwareParams/stabilizer/controller: 2
>  * /crazyswarm_server/firmwareParams/stabilizer/estimator: 2
>  * /crazyswarm_server/force_no_cache: False
>  * /crazyswarm_server/genericLogTopicFrequencies: [50, 50]
>  * /crazyswarm_server/genericLogTopic_log1_Variables: ['stateEstimate.x...
>  * /crazyswarm_server/genericLogTopic_log2_Variables: ['acc.x', 'acc.y'...
>  * /crazyswarm_server/genericLogTopics: ['log1', 'log2']
>  * /crazyswarm_server/motion_capture_type: vicon
>  * /crazyswarm_server/object_tracking_type: motionCapture
>  * /crazyswarm_server/print_latency: False
>  * /crazyswarm_server/save_point_clouds: ~/pointCloud.ot
>  * /crazyswarm_server/send_position_only: False
>  * /crazyswarm_server/vicon_host_name: 192.168.1.100:801
>  * /crazyswarm_server/world_frame: /world
>  * /crazyswarm_server/write_csvs: False
>  * /crazyswarm_teleop/csv_file: /home/coven-lab/c...
>  * /crazyswarm_teleop/timescale: 0.8
>  * /dynamicsConfigurations/0/maxFitnessScore: 0.001
>  * /dynamicsConfigurations/0/maxPitch: 1.4
>  * /dynamicsConfigurations/0/maxPitchRate: 20.0
>  * /dynamicsConfigurations/0/maxRoll: 1.4
>  * /dynamicsConfigurations/0/maxRollRate: 20.0
>  * /dynamicsConfigurations/0/maxXVelocity: 2.0
>  * /dynamicsConfigurations/0/maxYVelocity: 2.0
>  * /dynamicsConfigurations/0/maxYawRate: 10.0
>  * /dynamicsConfigurations/0/maxZVelocity: 3.0
>  * /joy/dev: /dev/input/js0
>  * /markerConfigurations/0/numPoints: 4
>  * /markerConfigurations/0/offset: [0.0, -0.01, -0.04]
>  * /markerConfigurations/0/points/0: [0.0177184, 0.013...
>  * /markerConfigurations/0/points/1: [-0.0262914, 0.05...
>  * /markerConfigurations/0/points/2: [-0.0328889, -0.0...
>  * /markerConfigurations/0/points/3: [0.0431307, -0.03...
>  * /markerConfigurations/1/numPoints: 4
>  * /markerConfigurations/1/offset: [0.0, 0.0, -0.03]
>  * /markerConfigurations/1/points/0: [-0.00896228, -0....
>  * /markerConfigurations/1/points/1: [-0.0156318, 0.09...
>  * /markerConfigurations/1/points/2: [0.0461693, -0.08...
>  * /markerConfigurations/1/points/3: [-0.0789959, -0.0...
>  * /markerConfigurations/2/numPoints: 4
>  * /markerConfigurations/2/offset: [0.0, 0.0, -0.06]
>  * /markerConfigurations/2/points/0: [0.0558163, -0.00...
>  * /markerConfigurations/2/points/1: [-0.0113941, 0.00...
>  * /markerConfigurations/2/points/2: [-0.0306277, 0.05...
>  * /markerConfigurations/2/points/3: [0.0535816, -0.04...
>  * /markerConfigurations/3/numPoints: 1
>  * /markerConfigurations/3/offset: [0.0, -0.01, -0.04]
>  * /markerConfigurations/3/points/0: [0.0177184, 0.013...
>  * /numDynamicsConfigurations: 1
>  * /numMarkerConfigurations: 4
>  * /rosdistro: melodic
>  * /rosversion: 1.14.9
> 
> NODES
>   /
>     crazyswarm_server (crazyswarm/crazyswarm_server)
>     crazyswarm_teleop (crazyswarm/crazyswarm_teleop)
>     joy (joy/joy_node)
>     rviz (rviz/rviz)
> 
> auto-starting new master
> process[master]: started with pid [29859]
> ROS_MASTER_URI=http://localhost:11311
> 
> setting /run_id to 00d7f46c-cfcd-11eb-868b-606dc7c60f33
> process[rosout-1]: started with pid [29870]
> started core service [/rosout]
> process[crazyswarm_server-2]: started with pid [29877]
> process[joy-3]: started with pid [29878]
> process[crazyswarm_teleop-4]: started with pid [29879]
> process[rviz-5]: started with pid [29889]
> [ INFO] [1623976495.980497579]: Wait for services...
> [ INFO] [1623976495.981646621]: waitForService: Service [/emergency] has not been advertised, waiting...
> [ INFO] [1623976496.023255639]: waitForService: Service [/emergency] is now available.
> [ INFO] [1623976496.024254839]: Manager ready.
> ch: 1
> [ INFO] [1623976496.048808364]: Adding CF: cf7 (radio://0/100/2M/E7E7E7E707, cf7)...
> [ INFO] [1623976496.077270910]: CF ctor: 0.027858 s
> [ INFO] [1623976496.081400659]: [cf7] ---
> [ INFO] [1623976496.083922698]: [cf7] SYS: Crazyflie 2.0 is up and running!
> [ INFO] [1623976496.086613432]: [cf7] SYS: Build 215:a7e0aae54b23 (2021.03 +215) MODIFIED
> [ INFO] [1623976496.090227326]: [cf7] SYS: I am 0x313139313535511600290050 and I have 1024KB of flash!
> [ INFO] [1623976496.091715917]: [cf7] CFGBLK: v1, verification [OK]
> [ INFO] [1623976496.093039683]: [cf7] DECK_CORE: 1 deck(s) found
> [ INFO] [1623976496.095702548]: [cf7] DECK_CORE: Calling INIT on driver bcUSD for deck 0
> [ INFO] [1623976496.097033799]: [cf7] uSD: mount SD-Card [OK].
> [ INFO] [1623976496.098260177]: [cf7] uSD: wait for sensors
> [ INFO] [1623976496.099748557]: [cf7] MPU9250 I2C connection [OK].
> [ INFO] [1623976496.101185653]: [cf7] AK8963 I2C connection [OK].
> [ INFO] [1623976496.102707728]: [cf7] LPS25H I2C connection [OK].
> [ INFO] [1623976496.105301162]: [cf7] ESTIMATOR: Using Complementary (1) estimator
> [ INFO] [1623976496.107073705]: [cf7] roller
> [ INFO] [1623976496.111438946]: [cf7] MTR-DRV: Using brushed motor dSYS: About to run tests in system.c.
> [ INFO] [1623976496.112892633]: [cf7] EEPROM: I2C connection [OK].
> [ INFO] [1623976496.114316281]: [cf7] STORAGE: Storage check [OK].
> [ INFO] [1623976496.115580915]: [cf7] AK8963: Self test [OK].
> [ INFO] [1623976496.116977779]: [cf7] DECK_CORE: Deck 0 test [OK].
> [ INFO] [1623976496.118227125]: [cf7] SYS: Self test passed!
> [ INFO] [1623976496.120498794]: [cf7] STAB: Wait for sensor calibration...
> [ INFO] [1623976496.122552443]: [cf7] STAB: Ready to fly.
> [ INFO] [1623976496.123899060]: [cf7] uSD: Config read [FAIL].
> [ INFO] [1623976496.126529751]: [cf7] STAB: WARNING: stabilizer loop rate is off (899)
> [ INFO] [1623976496.127380377]: [cf7] Requesting parameters...
> [ INFO] [1623976496.153820463]: Found variables in cache.
> [ WARN] [1623976496.162434158]: Couldn't set gain on joystick force feedback: Bad file descriptor
> [ INFO] [1623976496.163331717]: Opened joystick: /dev/input/js0. deadzone_: 0.050000.
> [ INFO] [1623976496.488757898]: [cf7] reqParamTOC: 0.411428 s (225 params)
> [ INFO] [1623976496.488791377]: [cf7] Requesting logging variables...
> [ INFO] [1623976496.491426027]: Found variables in cache.
> [ INFO] [1623976496.491923678]: [cf7] reqLogTOC: 0.003179 s
> [ INFO] [1623976496.504968320]: [cf7] logBlocks: 0.411428 s
> [ INFO] [1623976496.505001539]: Requesting memories...
> [ INFO] [1623976496.507455645]: Memories: 6
> [ INFO] [1623976496.526544293]: [cf7] Ready. Elapsed: 0.449209 s
> [ INFO] [1623976496.526583002]: CF run: 0.449341 s
> update commander/enHighLevel to 1
> update kalman/resetEstimation to 1
> update locSrv/extPosStdDev to 0.001
> update locSrv/extQuatStdDev to 0.05
> update ring/effect to 16
> update ring/headlightEnable to 0
> update ring/solidBlue to 255
> update ring/solidGreen to 0
> update ring/solidRed to 0
> update stabilizer/controller to 2
> update stabilizer/estimator to 2
> update ctrlMel/i_range_m_z to 1500
> update ctrlMel/i_range_xy to 2
> update ctrlMel/i_range_z to 0.4
> update ctrlMel/kR_xy to 70000
> update ctrlMel/kR_z to 60000
> update ctrlMel/kd_omega_rp to 200
> update ctrlMel/kd_xy to 0.2
> update ctrlMel/kd_z to 0.4
> update ctrlMel/ki_m_z to 500
> update ctrlMel/ki_xy to 0.05
> update ctrlMel/ki_z to 0.05
> update ctrlMel/kp_xy to 0.4
> update ctrlMel/kp_z to 1.25
> update ctrlMel/kw_xy to 20000
> update ctrlMel/kw_z to 12000
> update ctrlMel/mass to 0.032
> update ctrlMel/massThrust to 132000
> [ INFO] [1623976496.540961150]: [cf7] Update parameters
> [ INFO] [1623976496.569390670]: [cf7] CONTROLLER: Using Mellinger (2) controller
> [ INFO] [1623976496.573227564]: [cf7] ESTIMATOR: Using Kalman (2) estimator
> [ INFO] [1623976496.598462380]: Update params: 0.071794 s
> [ INFO] [1623976496.598867017]: Started 1 threads
> [ WARN] [1623976496.610836838]: Latency too high! Is 0.011532 s.
> [ WARN] [1623976508.098597569]: [cf7] Link Quality low (0.670000)
> [ INFO] [1623976518.085269276]: [cf7] Upload trajectory
> [ WARN] [1623976518.133226713]: Latency too high! Is 0.043123 s.
> [ WARN] [1623976518.182462248]: Latency too high! Is 0.049039 s.
> [ INFO] [1623976518.184947130]: [cf7] Uploaded trajectory
> [ INFO] [1623976518.195009090]: Takeoff!
> [ INFO] [1623976520.727574842]: [cf7] GoTo
> [ INFO] [1623976523.244872004]: Start trajectory!
> [ INFO] [1623976532.565733093]: Land!

**Reply by whoenig** [reply]
The latest version uses a different config file format, see https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/decks/micro-sd-card-deck/.
